osu.Game/Screens/Play/GameplayClockContainer.cs

```diff
@@ -154,13 +164,16 @@ public void Restart()
 
         public void Start()
         {
-            // Seeking the decoupled clock to its current time ensures that its source clock will be seeked to the same time
-            // This accounts for the audio clock source potentially taking time to enter a completely stopped state
-            Seek(GameplayClock.CurrentTime);
-            adjustableClock.Start();
-            IsPaused.Value = false;
+            if (!adjustableClock.IsRunning)
+            {
+                // Seeking the decoupled clock to its current time ensures that its source clock will be seeked to the same time
+                // This accounts for the audio clock source potentially taking time to enter a completely stopped state
+                Seek(GameplayClock.CurrentTime);
```

Comment:
Just to clarify that I understand the mode of failure correct: this "worked" when multi-threaded but failed single-threaded because in multi-threaded mode enqueued seeks could execute while the update thread was doing its things, making this a race in the multi-threaded case:

```
Update thread						Audio thread

enqueues seek to requested time
									executes enqueued 1st seek
enqueues seek to current time
									executes enqueued 2nd seek
```

and making the first seek a no-op in the single-threaded case (as the second seek reverted it entirely), right?

```
Frame N

update loop enqueues seek to requested time
update loop enqueues seek to current time

Frame N+1

audio thread executes enqueued 1st seek
audio thread executes enqueued 2nd seek <-- reverting 1st seek
```

Reply:
That's correct, yeah. The audio thread was running so often that it was not an issue in multi-threaded execution.